test/ResponseFactoryTest.php

```diff
@@ -2,19 +2,42 @@
 
 namespace Interop\Http\Factory;
 
-final class ResponseFactoryTest extends ResponseFactoryTestCase
-{
-    /**
-     * {@inheritdoc}
-     */
-    protected function createResponseFactory()
-    {
-        if (!defined('RESPONSE_FACTORY')) {
-            $this->markTestSkipped('Response factory class name not provided');
-        }
-
-        $factoryClass = RESPONSE_FACTORY;
-
-        return new $factoryClass();
-    }
+use PHPUnit\Framework\Attributes\DataProvider;
+use PHPUnit\Framework\TestCase;
+use Psr\Http\Message\ResponseFactoryInterface;
+use function class_exists;
+use function defined;
+
+class ResponseFactoryTest extends TestCase{
+
+	protected ResponseFactoryInterface $responseFactory;
+
+	/**
+	 * @noinspection PhpUndefinedConstantInspection
+	 */
+	public function setUp():void{
+
+		if(!defined('RESPONSE_FACTORY') || !class_exists(RESPONSE_FACTORY)){
+			static::markTestSkipped('RESPONSE_FACTORY class name not provided');
+		}
+
+		$this->responseFactory = new (RESPONSE_FACTORY);
```

Comment:
This change is out of scope. Please revert it. Same goes for the rest of this PR.

The only changes should be to composer.json and anything that is necessary for PHPUnit to not give warnings about deprecations.